src/main/resources/GoogleApps/GoogleAppsConfigSheet.xml

```diff
@@ -43,13 +42,92 @@
   <syntaxId>xwiki/2.0</syntaxId>
   <hidden>true</hidden>
   <content>{{velocity}}
-## You can modify this page to customize the presentation of your object.
-## At first you should keep the default presentation and just save the document.
+$xwiki.jsx.use('GoogleApps.GoogleAppsConfigClass')
+$xwiki.ssx.use('GoogleApps.GoogleAppsConfigClass')
 
-#set($class = $doc.getObject('GoogleApps.GoogleAppsConfigClass').xWikiClass)
-#foreach($prop in $class.properties)
-  ; $prop.prettyName
-  : $doc.display($prop.getName())
+#set($configClassName="GoogleApps.GoogleAppsConfig")
+#set($formId = "${section.toLowerCase()}_${configClassName}")
+#set($configDoc = $xwiki.getDocument($configClassName))
+#set($className="GoogleApps.GoogleAppsConfigClass")
+#set($prefix="${configDoc.fullName}_${className}_0")
+#set($obj = $configDoc.getObject($className))
+#macro(t $n)##
+    $services.localization.render("GoogleApps.GoogleAppsConfigClass_${n}")##
+#end
+#macro(tp $n $p)##
+    $services.localization.render("GoogleApps.GoogleAppsConfigClass_${n}",$p)##
 #end
-{{/velocity}}</content>
+
+{{html wiki=true}}
+&lt;form id="$formId" method="post" action="$xwiki.getURL($configDoc, 'saveandcontinue')" onsubmit="cancelCancelEdit()" class="xform"&gt;
+    ; $configDoc.display("activate", 'edit', $obj) &lt;label for="${prefix}_activate"&gt;#t("activate")&lt;/label&gt;
+
+    &lt;fieldset&gt;
+        &lt;legend&gt;#t("communicate")&lt;/legend&gt;
+
+        ; &lt;span class="xHint"&gt;##
+           #tp("communicate.hint",  ['[[', '&gt;&gt;https://accounts.google.com/ManageDomains]]'])&lt;br&gt;
```

Comment:
I believe that this would make the sheet code look quite unreadable.
See how this is there in the JS-extension for just the two lines where we need it from velocity; bloated long lines and hard to keep the overview.

Without a solution to shorten the key prefix, I think that it needs to be accepted.
Is shortening a key prefix an option?

paul